tests/parsers/test_pw.py

```diff
@@ -907,3 +907,35 @@ def test_pw_vcrelax_failed_not_converged_nstep(
     assert 'output_parameters' in results
     assert 'output_structure' in results
     assert 'output_trajectory' in results
+
+
+def test_environ(
+    fixture_localhost,
+    generate_calc_job_node,
+    generate_parser,
+    generate_inputs,
+    data_regression,
+):
+    """Test a simple Environ calculation."""
+    name = 'default'
```

Comment:
You are almost there with creating a parser test. This `name` variable here, actually corresponds to a folder in the `tests/parsers/fixtures/pw` folder. You will find the `default` folder there. It actually contains output files that were generated by a `PwCalculation`, i.e. the `aiida.out` and `data-file.xml`. Through the `generate_calc_job_node` function called on line 924, a `CalcJobNode` is mocked with a `retrieved` output node that contains these files. This is replicating what happens when you actually run a `PwCalculation` and the node is passed to the parser.

What you then have to do is create a new folder in that fixtures base folder, let's call it `default_environ` and put the `aiida.out` and `data-file.xml` in there that you take from an actual `PwCalculation` run that included the `ENVIRON` namecard. Only then will you actually test the parser for output files that were produced by an environ run. In the current state, you are simply testing exactly the same as the `test_default` test, which tests the outputs for a simple scf calculation with `pw.x`. I hope this is clear. If not, feel to ask for more clarification.

Reply:
Yup all makes sense - thanks! Let me just describe what I did here. I added a new structure with `structure_id=platinum` to the `conftest.py` in order to test the different environ outputs as they won't work for the two example cases for Si or water (I need a 2D slab to check for the right flags). I then ran an `pw` calculation with QE v.6.7 and stored the `aiida.out` and the `data-file.xml` file in `fixtures/default_environ`. So far so good, except the auto-generated `test_environ.yml` file doesn't seem to have the right specifications (it reports Si and not Pt). Moreover the `k-points` do not seem to be parsed properly (commented out currently). I can also bypass this and run everything for Si, but my thinking is that it would be nice to have the test mirror that of the Environ package.